### PR TITLE
ci-operator: label all resources it creates

### PR DESCRIPTION
**Pull request by Prucek**
This PR adds:
- LabeledClient wrapper for labeling all resources ci-operator creates
- ProwJob information labels to the already existing labels

Resolves: [DPTP-4117](https://issues.redhat.com/browse/DPTP-4117)

### CODE REVIEW

File: pkg/labeledclient/client.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	return c.upstream.IsObjectNamespaced(obj)
+}
+
+func (c *client) SubResource(subResource string) ctrlruntimeclient.SubResourceClient {
+	return c.upstream.SubResource(subResource)
+}
+
+func (c *client) addLabels(obj ctrlruntimeclient.Object) {
+	// SelfSubjectAccessReview & LocalSubjectAccessReview does not hold labels
+	_, ok := obj.(*authapi.SelfSubjectAccessReview)
```
[the remark this answers is not in the document]
Comment (droslean) [reply pkg/labeledclient/client.go:97]: I don't know :)... Let's add those we are creating....

[the remark this answers is not in the document]
Comment (Prucek) [reply pkg/labeledclient/client.go]: we want to label all resources created by ci-operator, these are the only ones that can't hold labels

File: pkg/labeledclient/client.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+func (c *client) IsObjectNamespaced(obj runtime.Object) (bool, error) {
+	return c.upstream.IsObjectNamespaced(obj)
+}
+
+func (c *client) SubResource(subResource string) ctrlruntimeclient.SubResourceClient {
+	return c.upstream.SubResource(subResource)
+}
+
+func (c *client) addLabels(obj ctrlruntimeclient.Object) {
```
Comment (droslean) [inline pkg/labeledclient/client.go:95]:
```go
func (c *client) addLabels(obj ctrlruntimeclient.Object) {
    switch obj.(type) {
    case *authapi.SelfSubjectAccessReview, *authapi.LocalSubjectAccessReview:
    default:
        obj.SetLabels(steps.LabelsFor(c.jobSpec, obj.GetLabels(), ""))
    }
}
```

**Comment by droslean** [comment]
@Prucek I left a nit comment [here](https://github.com/openshift/ci-tools/pull/4306/files#r1752047360), let's merge this to unblock your work. Feel free to address the nit in a follow-up

/lgtm
/approve

File: pkg/labeledclient/client.go (hunk 2 above)
Comment (Prucek) [reply pkg/labeledclient/client.go:95]: I already did that :D

**Pull request merged**